kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -399,45 +399,47 @@ BatchedInternalNode DBAdapter::Reader::get_internal(const InternalNodeKey &key)
   return deserialize<BatchedInternalNode>(res);
 }
 
-Status DBAdapter::addLastReachableBlock(const SetOfKeyValuePairs &updates) {
+BlockId DBAdapter::addBlock(const SetOfKeyValuePairs &updates) {
   for (const auto &kv : updates) {
     if (kv.second.empty()) {
       const auto msg = "Adding empty values in a block is not supported";
       LOG_ERROR(logger_, msg);
```

Comment:
I think we don't need to log error twice - it's logged in place it's caught

Reply:
I think you are right. However, if there is a bug and there is a missing exception handler or a missing log call in the handler, we might miss the event. Additionally, since these are errors and won't happen often it wouldn't be a big burden on the log. I will leave them for now and if we have an issue, we can easily remove.